libtiledbsoma/src/soma/managed_query.h

```diff
@@ -614,11 +614,17 @@ class ManagedQuery {
         // additional processing steps
 
         UserType* buf;
+        // The array->offset is non-zero when we are passed sliced
+        // Arrow tables like arrow_table[:m] or arrow_table[m:].
         if (array->n_buffers == 3) {
             buf = (UserType*)array->buffers[2] + array->offset;
         } else {
             buf = (UserType*)array->buffers[1] + array->offset;
         }
+        uint8_t* validity = (uint8_t*)array->buffers[0];
```

Comment:
Nit: This would be a good opportunity to move away from the C-style cast to a named cast (`static_cast` in this case).